stacks_queues/queue.py

```diff
@@ -12,50 +22,76 @@ class Queue:
     def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
-        self.front = -1
+        self.front = 0
         self.rear = -1
         self.size = 0
-      
-
+    
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException('This will break it')
+        
+        # the last element 
+        self.rear = (self.rear +1) % self.buffer_size
+        self.store[self.rear] = element
+        self.size = self.size +1
+
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        #  find element in the store
+        # First in first out 
+        if self.size == 0:
+            raise QueueEmptyException('This will break it')
+        else:
+            val = self.store[self.front]
+            self.front = self.front + 1 % self.buffer_size
+        
+        # decrease size
+        self.size = self.size - 1
+        return val
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
-        
+        if self.store[self.front]:
+            return self.store[0]
```

Comment:
⚠️ With a circular queue, the front of the queue may not always be at index 0. That's why we have a front pointer that shifts as elements are dequeued.

The current tests don't cover `front` which is why you  may not have caught this.
```suggestion
            return self.store[self.front]
```